### PR TITLE
[#2106] Made sure right clicking on a currently dragged trackevent with ...

### PR DESCRIPTION
**Pull request by dseif**
...a ghost doesn't break

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/37992fc42492727/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/37992fc42492727/output.txt

Total script time: 1.00 mins

Server is running on http://pr951.butyr.org/templates/basic/
Run the Test Suite: http://pr951.butyr.org/test/

File: src/util/dragndrop.js [hunk 1]
```diff
@@ -87,6 +87,7 @@ define([], function(){
   function onMouseUp( e ){
     __mouseDown = false;
     window.removeEventListener( "mousemove", onDragged, false );
+    window.removeEventListener( "mousedown", onMouseUp, false );
```
Comment (secretrobotron) [inline src/util/dragndrop.js]: This can't be right.

File: src/timeline/track-container.js [hunk 2]
```diff
@@ -109,7 +109,10 @@ define( [ "core/logger", "util/dragndrop", "./trackevent-drag-manager" ],
       var trackEventView = e.target,
           element = trackEventView.element,
           trackView = trackEventView.trackEvent.track.view;
-      trackView.element.removeChild( element );
+
+      if ( trackView.element === element.parentNode ) {
+        trackView.element.removeChild( element );
+      }
```
Comment (secretrobotron) [inline src/timeline/track-container.js]: Does this check need to happen with the changes you made to dragndrop?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_cleanup** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/6e7e1f9da3ee2da/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/6e7e1f9da3ee2da/output.txt

Total script time: 1.23 mins

**Pull request closed without merging**